src/app/contexts/UserContext/index.tsx

```diff
@@ -19,10 +23,25 @@ export const UserContext = React.createContext<UserContextProps>(
   {} as UserContextProps,
 );
 
+const cknsMvtCookie = () => {
+  const cookieName = 'ckns_mvt';
+  const cookieValue = Cookie.get(cookieName);
+
+  if (!cookieValue) {
+    const cookieUuid = uuid();
+    const expires = 365;
+    Cookie.set(cookieName, cookieUuid, { expires, path: '/', secure: true });
+  }
+};
+
 export const UserContextProvider = ({ children }: PropsWithChildren) => {
   const [cookiePolicy, setCookiePolicy] = useState(getCookiePolicy());
   const [chartbeatConfig, sendCanonicalChartbeatBeacon] = useState(null);
 
+  if (onClient() && !isOperaProxy()) {
```

Comment:
does this mean that we dont set the cookie for opera mini (based of this and the unit tests)? can you remind me why if so! assuming that we just dont run experiments on opera mini for a weird opera reason

Reply:
Actually debated wether to set this on opera mini or not. The reason is because Opera mini extreme mode renders it's pages for the user on their servers, so that means all users using Opera Mini Extreme mode would have the same userId, making testing redundant. Although technically we can set it here as [this](https://github.com/bbc/simorgh/blob/84ab860d71fa7dc366c9da659484b80426996d39/src/app/legacy/containers/PageHandlers/withOptimizelyProvider/index.jsx#L36) effectively disable running optimizely for opera mini users anyway, I'll ask product and @andrewscfc to see what they think